endaq/calc/shock.py

```diff
@@ -11,71 +12,247 @@
 import pandas as pd
 import scipy.signal
 
+from endaq.calc.stats import L2_norm
+from endaq.calc import utils
 
-def rel_displ(df: pd.DataFrame, omega: float, damp: float = 0) -> pd.DataFrame:
-    """Calculate the relative displacement for a SDOF system."""
-    # Generate the transfer function
-    #   H(s) = L{z(t)}(s) / L{y"(t)}(s) = (1/s²)(Z(s)/Y(s))
-    # for the PDE
-    #   z" + (2ζω)z' + (ω^2)z = -y"
-    dt = (df.index[-1] - df.index[0]) / (len(df.index) - 1)
-    if isinstance(dt, (np.timedelta64, pd.Timedelta)):
-        dt = dt / np.timedelta64(1, "s")
 
+def _rel_displ_transfer_func(omega: float, damp: float = 0.0, dt: float = 1.0):
```

Comment:
Type Hint for return

Reply:
see 06f2b0ae993894190bbe6c5aa510cddf75c58bde